web/src/main/java/org/springframework/security/web/authentication/UsernamePasswordAuthenticationFilter.java

```diff
@@ -74,10 +74,10 @@ public Authentication attemptAuthentication(HttpServletRequest request, HttpServ
 		if (this.postOnly && !request.getMethod().equals("POST")) {
 			throw new AuthenticationServiceException("Authentication method not supported: " + request.getMethod());
 		}
+
 		String username = obtainUsername(request);
-		username = (username != null) ? username.trim() : "";
 		String password = obtainPassword(request);
-		password = (password != null) ? password : "";
```

Comment:
For backward compatibility, this line needs to stay where it is. If someone is overriding obtainPassword today, then this change would mean when they upgrade, the defaulting would stop.